core/base-service/redirector.js

```diff
@@ -15,18 +16,19 @@ const trace = require('./trace')
 const attrSchema = Joi.object({
   category: isValidCategory,
   route: isValidRoute,
-  target: Joi.func()
+  transformPath: Joi.func()
     .maxArity(1)
     .required()
     .error(
       () =>
-        '"target" must be a function that transforms named params to a new path'
+        '"transformPath" must be a function that transforms named params to a new path'
     ),
+  transformQueryParams: Joi.func().arity(1),
```

Comment:
Given the sole purpose of this function (transforming a route param to a query param, https://github.com/badges/shields/issues/3118#issuecomment-468503034) I believe it makes sense to require the argument be passed to the function